connection.go

```diff
@@ -259,8 +259,8 @@ func newConnection(connString string) (*connection, error) {
 	// Read SSL/TLS flag.
 	sslFlag := strings.ToLower(result.connURL.Query().Get("tlsmode"))
 	if sslFlag == "" {
-		// Set default to tlsModeServerPrefer
-		sslFlag = tlsModeServerPrefer
+		// Set default to tlsModePrefer
+		sslFlag = tlsModeNone
```

Comment:
I changed this back to 'tlsModeNone' as it was initially which is an appropriate logic. This change made all the workflow tests pass.